tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -2354,7 +2386,11 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
         value match {
           case js.Function(args, body) =>
             // Make sure the function has a meaningful `name` property
-            js.FunctionDef(globalVarIdent, args, body)
+            val functionExpr = js.FunctionDef(globalVarIdent, args, body)
+            if (keepFunctionExpression)
+              js.VarDef(globalVarIdent, Some(functionExpr))
```

Comment:
It caused a `ParseError` in PhantomJS if I did that differently. I'm not sure there's something against that in the ES spec (5.1, 5.1 strict and/or 2015), but I seem to remember I saw something about that someday. The current implementation is safe and assured to always work.